### PR TITLE
Adds game updating with computer player moves

### PR DESCRIPTION
**Pull request by samhandev**
@jsuchy 
@marosluuce 

Implement code for the game to update with computer player moves.

### CODE REVIEW

File: lib/tic_tac_toe_web/game_updater.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    def next_game_state
+      while game.running? && move_available?
+        game.make_move(current_move)
+      end
+      game
+    end
+
+    private
+
+    def move_available?
```
Comment (samhandev) [inline lib/tic_tac_toe_web/game_updater.rb:16]: Rename some things here to make it clear what's actually going on. Really you're asking if it's a human player. Don't violate OCP, but find a way to make this more obvious.

File: lib/tic_tac_toe_web/game_updater.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      @current_move = game.get_current_player_move
+      if current_move
```
Comment (hanster) [inline lib/tic_tac_toe_web/game_updater.rb]: !current_move.nil?

File: lib/tic_tac_toe_web/models/game.rb [hunk 3]
```diff
@@ -12,7 +14,7 @@ class Game
       CLASS_SIZE_4 = "size_four"
 
       def initialize(game, player_type, board_type)
-        @game = game
+        @game = GameUpdater.new(game).next_game_state
```
Comment (samhandev) [inline lib/tic_tac_toe_web/models/game.rb:17]: This means we can't actually visualise the plays in a CvC game.
@game = next_state is a bit weird.

File: spec/tic_tac_toe_web/game_updater_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+require 'tic_tac_toe_web/game_updater'
+
+RSpec.describe TicTacToeWeb::GameUpdater do
+  let(:board) { instance_double('Board') }
+  let(:players) { [] }
+  let(:ui) { instance_double('UI') }
+
+  it "doesn't do any updates if the current player doesn't have a move" do
+    game = object_double(TicTacToeCore::Game.new(board, players, ui),
```
Comment (samhandev) [inline spec/tic_tac_toe_web/game_updater_spec.rb:10]: instance_double is simpler.

File: spec/tic_tac_toe_web/models/game_spec.rb [hunk 5]
```diff
@@ -13,6 +13,10 @@ module Models
       let(:game) { TicTacToeCore::Game.new(board, players, ui) }
       let(:game_model) { Game.new(game, 1, 3) }
 
+      before(:each) do
+        allow(game).to receive(:get_current_player_move).and_return(nil)
```
Comment (samhandev) [inline spec/tic_tac_toe_web/models/game_spec.rb:17]: smelly, don't let nils leak into your system.

**Pull request closed without merging**